Sources/UnionNotifications/iOS26NotificationView.swift

```diff
@@ -1,71 +1,133 @@
-                    .buttonStyle(.glass)
-                    .disabled(true)
-
-                    Button {
-                        onAllow()
-                    } label: {
-                        Text("Allow", bundle: .module)
-                            .font(.body.weight(.semibold))
-                            .foregroundStyle(.white)
-                            .frame(maxWidth: .infinity)
-                            .frame(height: 34)
-                    }
-                    .buttonStyle(.glassProminent)
+        GeometryReader { geometry in
+            GlassEffectContainer(spacing: 0) {
+                VStack(spacing: 0) {
+                    iOS26NotificationContent(
+                        appName: appName,
+                        shouldEnableScroll: shouldEnableScroll,
+                        maxContentHeight: NotificationDialogConstants.maxScrollHeight
+                    )
+                    
+                    buttonSection
                 }
-                .padding(.horizontal, 16)
             }
-            .padding(.top, 22)
-            .padding(.bottom, 16)
+            .frame(width: dialogWidth(proxy: geometry))
+            .glassEffect(.regular, in: .rect(cornerRadius: 32))
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+        }
+    }
+    
+    private var buttonSection: some View {
+        Group {
+            if shouldStackButtons {
+                verticalButtons
+            } else {
+                horizontalButtons
+            }
+        }
+        .padding(.horizontal, 16)
+        .padding(.bottom, 16)
+    }
+    
+    private var verticalButtons: some View {
+        VStack(spacing: 8) {
+            disabledButton
+            activeButton
+        }
+    }
+    
+    private var horizontalButtons: some View {
+        HStack(spacing: 8) {
+            disabledButton
+            activeButton
+        }
+    }
+    
+    @Environment(\.colorScheme) private var colorScheme
+    
+    private var disabledButton: some View {
+        Button(action: { }) {
+            Text("Don't Allow", bundle: .module)
+                .font(.body.weight(.medium))
+                .foregroundStyle(.secondary)
+                .frame(maxWidth: .infinity, minHeight: buttonHeight)
+        }
+        .background(.gray.opacity(0.15), in: .rect(cornerRadius: 30, style: .continuous))
+        .glassEffect(.clear, in: .rect(cornerRadius: 30))
+        .disabled(true)
+    }
+    
+
+    private var activeButton: some View {
+        Button(action: onAllow) {
+            Text("Allow", bundle: .module)
+                .font(.body.weight(.semibold))
+                .foregroundStyle(colorScheme == .dark ? .white : .black)
+                .frame(maxWidth: .infinity, minHeight: buttonHeight)
         }
-        .frame(width: 320)
-        .glassEffect(.regular, in: .rect(cornerRadius: 32))
+        .background(.gray.opacity(0.15), in: .rect(cornerRadius: 30, style: .continuous))
```

Comment:
How sure are you about this gray opacity effect? And how did you identify this? I would be pretty surprised if Apple were using a gray overlay combined with a clear effect rather than a regular glass effect for a system dialog. But if you tested extensively then no problem.